contracts/core/Safe.sol

```diff
@@ -48,5 +48,27 @@ contract Safe {
     return createSafe(ownerArr, 1);
   }
 
+  function create2Safe(address [] memory safeOwners, uint256 threshold, uint256 nonce) internal returns (address){
```

Comment:
@habdelra i was not too clear how import of this Safe module works with upgradeable contracts. would this behave nicely by adding this function?